api/external_auth.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto3";
+
+// [#proto-status: draft]
+
+package envoy.api.v2;
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+// Authorization service.
+// Primarily responds with OK or NOT OK.
+service Authorization {
+  rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+  // The name of the tenant for multi-tenant support.
+  string tenant_name = 1;
```

Comment:
Where does this come from today? Envoy doesn't have an explicit notion of multi-tenancy, even though it can be used in this capacity. For Google internal requirements, we encode this in xDS opaque metadata fields, so Envoy is never explicitly aware. I think we should file a feature request and add in this notion if we are going to make it first-class in the attributes here.

Reply:
Let's drop this for now and treat the PR to add it as a first-class attribute separately.